### ISSUE TITLE
An error ocurred while starting the kernel

### ISSUE THREAD

**Issue by mtanika** [spyder-ide/spyder#15787]
### What steps will reproduce the problem?

I have installed Python 3.9.2 and wanted to have Sypeder 5.0.3 to run it as the Interpreter.

Tools > Preferences > Python interpreter

When I restarted Spyder, I got the error in the title, plus the following msg:

---------------------
Your Python environment or installation doesn't have the spyder‑kernels module or the right version of it installed (>= 2.0.1 and < 2.1.0). Without this module is not possible for Spyder to create a console for you.

You can install it by running in a system terminal:

conda install spyder‑kernels=2.0

or

pip install spyder‑kernels==2.0.*
---------------------

I used the recommended update command from a command line:

pip install spyder‑kernels==2.0.*

and it seemed to have installed properly.

But ... after I restarted Spyder, I got the same message ... as if the installation was not effected ...

Appreciate any help on the upgrade.

Regards,
Mits
mtanika@gmail.com

=======================================

trying to run the install again:

C:\Program Files>pip install spyder-kernels==2.0.*
Requirement already satisfied: spyder-kernels==2.0.* in c:\users\mits_ltp\appdata\roaming\python\python39\site-packages (2.0.3)
Requirement already satisfied: jupyter-client>=5.3.4 in c:\users\mits_ltp\appdata\roaming\python\python39\site-packages (from spyder-kernels==2.0.*) (6.1.12)
Requirement already satisfied: ipython>=7.6.0 in c:\users\mits_ltp\appdata\roaming\python\python39\site-packages (from spyder-kernels==2.0.*) (7.24.1)
Requirement already satisfied: cloudpickle in c:\users\mits_ltp\appdata\roaming\python\python39\site-packages (from spyder-kernels==2.0.*) (1.6.0)
Requirement already satisfied: pyzmq>=17 in c:\users\mits_ltp\appdata\roaming\python\python39\site-packages (from spyder-kernels==2.0.*) (22.1.0)
Requirement already satisfied: decorator<5 in c:\users\mits_ltp\appdata\roaming\python\python39\site-packages (from spyder-kernels==2.0.*) (4.4.2)
Requirement already satisfied: ipykernel>=5.3.0 in c:\users\mits_ltp\appdata\roaming\python\python39\site-packages (from spyder-kernels==2.0.*) (5.5.5)
Requirement already satisfied: tornado>=4.2 in c:\users\mits_ltp\appdata\roaming\python\python39\site-packages (from ipykernel>=5.3.0->spyder-kernels==2.0.*) (6.1)
Requirement already satisfied: traitlets>=4.1.0 in c:\users\mits_ltp\appdata\roaming\python\python39\site-packages (from ipykernel>=5.3.0->spyder-kernels==2.0.*) (5.0.5)
Requirement already satisfied: pygments in c:\users\mits_ltp\appdata\roaming\python\python39\site-packages (from ipython>=7.6.0->spyder-kernels==2.0.*) (2.9.0)
Requirement already satisfied: matplotlib-inline in c:\users\mits_ltp\appdata\roaming\python\python39\site-packages (from ipython>=7.6.0->spyder-kernels==2.0.*) (0.1.2)
Requirement already satisfied: jedi>=0.16 in c:\users\mits_ltp\appdata\roaming\python\python39\site-packages (from ipython>=7.6.0->spyder-kernels==2.0.*) (0.18.0)
Requirement already satisfied: backcall in c:\users\mits_ltp\appdata\roaming\python\python39\site-packages (from ipython>=7.6.0->spyder-kernels==2.0.*) (0.2.0)
Requirement already satisfied: setuptools>=18.5 in c:\program files\python39\lib\site-packages (from ipython>=7.6.0->spyder-kernels==2.0.*) (49.2.1)
Requirement already satisfied: pickleshare in c:\users\mits_ltp\appdata\roaming\python\python39\site-packages (from ipython>=7.6.0->spyder-kernels==2.0.*) (0.7.5)
Requirement already satisfied: colorama in c:\users\mits_ltp\appdata\roaming\python\python39\site-packages (from ipython>=7.6.0->spyder-kernels==2.0.*) (0.4.4)
Requirement already satisfied: prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0 in c:\users\mits_ltp\appdata\roaming\python\python39\site-packages (from ipython>=7.6.0->spyder-kernels==2.0.*) (3.0.18)
Requirement already satisfied: parso<0.9.0,>=0.8.0 in c:\users\mits_ltp\appdata\roaming\python\python39\site-packages (from jedi>=0.16->ipython>=7.6.0->spyder-kernels==2.0.*) (0.8.2)
Requirement already satisfied: jupyter-core>=4.6.0 in c:\users\mits_ltp\appdata\roaming\python\python39\site-packages (from jupyter-client>=5.3.4->spyder-kernels==2.0.*) (4.7.1)
Requirement already satisfied: python-dateutil>=2.1 in c:\users\mits_ltp\appdata\roaming\python\python39\site-packages (from jupyter-client>=5.3.4->spyder-kernels==2.0.*) (2.8.1)
Requirement already satisfied: pywin32>=1.0 in c:\users\mits_ltp\appdata\roaming\python\python39\site-packages (from jupyter-core>=4.6.0->jupyter-client>=5.3.4->spyder-kernels==2.0.*) (301)
Requirement already satisfied: wcwidth in c:\users\mits_ltp\appdata\roaming\python\python39\site-packages (from prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0->ipython>=7.6.0->spyder-kernels==2.0.*) (0.2.5)
Requirement already satisfied: six>=1.5 in c:\users\mits_ltp\appdata\roaming\python\python39\site-packages (from python-dateutil>=2.1->jupyter-client>=5.3.4->spyder-kernels==2.0.*) (1.16.0)
Requirement already satisfied: ipython-genutils in c:\users\mits_ltp\appdata\roaming\python\python39\site-packages (from traitlets>=4.1.0->ipykernel>=5.3.0->spyder-kernels==2.0.*) (0.2.0)

### Traceback
```python-traceback
Traceback (most recent call last):
  File "C:\Program Files\Spyder\pkgs\spyder\plugins\ipythonconsole\plugin.py", line 1309, in close_client
    client.close()
  File "C:\Program Files\Spyder\pkgs\spyder\plugins\ipythonconsole\widgets\client.py", line 577, in close
    self.get_kernel() is None or self.slave)
  File "C:\Program Files\Spyder\pkgs\spyder\plugins\ipythonconsole\widgets\shell.py", line 164, in will_close
    self.spyder_kernel_comm.close()
  File "C:\Program Files\Spyder\pkgs\spyder\plugins\ipythonconsole\comms\kernelcomm.py", line 144, in close
    self.shutdown_comm_channel()
  File "C:\Program Files\Spyder\pkgs\spyder\plugins\ipythonconsole\comms\kernelcomm.py", line 87, in shutdown_comm_channel
    channel = self.kernel_client.comm_channel
AttributeError: 'NoneType' object has no attribute 'comm_channel'
```

## Versions

* Spyder version: 5.0.3 
* Python version: 3.7.9
* Qt version: 5.12.10
* PyQt5 version: 5.12.3
* Operating System: Windows 10

### Dependencies

```

# Mandatory:
atomicwrites >=1.2.0          :  1.4.0 (OK)
chardet >=2.0.0               :  4.0.0 (OK)
cloudpickle >=0.5.0           :  1.6.0 (OK)
cookiecutter >=1.6.0          :  1.7.3 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree                  :  None (OK)
IPython >=7.6.0               :  7.23.1 (OK)
jedi =0.17.2                  :  0.17.2 (OK)
jsonschema >=3.2.0            :  3.2.0 (OK)
keyring >=17.0.0              :  23.0.1 (OK)
nbconvert >=4.0               :  6.0.7 (OK)
numpydoc >=0.6.0              :  1.1.0 (OK)
paramiko >=2.4.0              :  2.7.2 (OK)
parso =0.7.0                  :  0.7.0 (OK)
pexpect >=4.4.0               :  4.8.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.8.0 (OK)
pygments >=2.0                :  2.9.0 (OK)
pylint >=1.0                  :  2.8.2 (OK)
pyls >=0.36.2;<1.0.0          :  0.36.2 (OK)
pyls_black >=0.4.6            :  0.4.6 (OK)
pyls_spyder >=0.3.2;<0.4.0    :  0.3.2 (OK)
qdarkstyle =3.0.2             :  3.0.2 (OK)
qstylizer >=0.1.10            :  0.2.0 (OK)
qtawesome >=1.0.2             :  1.0.2 (OK)
qtconsole >=5.1.0             :  5.1.0 (OK)
qtpy >=1.5.0                  :  1.9.0 (OK)
rtree >=0.9.7                 :  0.9.7 (OK)
setuptools >=39.0.0           :  56.2.0 (OK)
sphinx >=0.6.6                :  3.5.4 (OK)
spyder_kernels >=2.0.3;<2.1.0 :  2.0.3 (OK)
textdistance >=4.2.0          :  4.2.1 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog                      :  1.0.2 (OK)
zmq >=17                      :  22.0.3 (OK)

# Optional:
cython >=0.21                 :  0.29.23 (OK)
matplotlib >=2.0.0            :  3.4.2 (OK)
numpy >=1.7                   :  1.19.3 (OK)
pandas >=1.1.1                :  1.2.4 (OK)
scipy >=0.17.0                :  1.6.3 (OK)
sympy >=0.7.3                 :  1.8 (OK)
```

**Comment by juanis2112** [comment]
Hi @mtanika, this bug was already fixed and the fix will be available in our next version (5.0.4), to be released shortly. :)

**Comment by mtanika** [comment]
Gracias, Juanita !
